### ISSUE TITLE
Remove dependencies on k8s.io/kubernetes

### ISSUE THREAD

**Issue by msau42** [kubernetes-csi/external-provisioner#1067]
**What happened**:
Packages from k8s.io/kubernetes were not designed to be imported and adds a lot of complexity to our dependency management where we need to inject "replace" statements into go.mod. It is also not supported by dependabot, which we would like to use to help automate our releases.

It looks like the dependencies are all coming from the e2e tests, which are importing the kubernetes e2e test framework libraries:
https://github.com/search?q=repo%3Akubernetes-csi%2Fexternal-provisioner+k8s.io%2Fkubernetes+path%3A*.go&type=code

@xing-yang I think we may want to move this out to its own repo, or maybe into the csi-test repo?

**Comment by xing-yang** [comment]
@msau42 Right, we need to think about this.  Maybe csi-test repo is a better choice rather than creating another repo.

**Comment by xing-yang** [comment]
cc @RaunakShah

**Comment by msau42** [comment]
Another approach is to move the e2e test directory into its own go module with it's own go.mod.

This will help with CVE patching because dependabot today cannot support all the replace statements we have to do as a result of importing k8s.io/kubernetes.
